js/modules/k6/experimental/tracing/client_test.go

```diff
@@ -144,6 +149,105 @@ func TestClientInstrumentArguments(t *testing.T) {
+	_, hasTraceIDKey := testCase.client.vu.State().Tags.GetCurrentValues().Metadata["trace_id"]
+	assert.False(t, hasTraceIDKey)
+
+	// The callFn will assert that the trace_id key is present in vu metadata
+	// before returning
+	_ = testCase.client.instrumentedCall(callFn)
+
+	// Assert there is no trace_id key in vu metadata after using intrumentedCall
+	_, hasTraceIDKey = testCase.client.vu.State().Tags.GetCurrentValues().Metadata["trace_id"]
+	assert.False(t, hasTraceIDKey)
+}
+
+// This test ensures that the trace_id is added to the vu metadata when
+// and instrumented request is called; and that we can find it in the
+// produced samples.
+//
+// It also ensures that the trace_id is removed from the vu metadata
+// after the request is done.
+func TestCallingInstrumentedRequestEmitsTraceIdMetadata(t *testing.T) {
+	t.Parallel()
+
+	testCase := newTestSetup(t)
+	rt := testCase.TestRuntime.VU.Runtime()
+
+	// Making sure the instrumentHTTP is called in the init context
+	// before the test.
+	_, err := rt.RunString(`
+		let http = require('k6/http')
+		instrumentHTTP({propagator: 'w3c'})
+	`)
+	require.NoError(t, err)
+
+	// Move to VU context. Setup in way that the produced samples are
+	// written to a channel we own.
+	samples := make(chan metrics.SampleContainer, 1000)
+	httpBin := httpmultibin.NewHTTPMultiBin(t)
+	testCase.TestRuntime.MoveToVUContext(&lib.State{
+		BuiltinMetrics: metrics.RegisterBuiltinMetrics(testCase.TestRuntime.VU.InitEnvField.Registry),
+		Tags:           lib.NewVUStateTags(testCase.TestRuntime.VU.InitEnvField.Registry.RootTagSet()),
+		Transport:      httpBin.HTTPTransport,
+		BPool:          bpool.NewBufferPool(1),
+		Samples:        samples,
+		Options:        lib.Options{SystemTags: &metrics.DefaultSystemTagSet},
+	})
+
+	// Inject a function in the JS runtime to assert the trace_id key
+	// is present in the vu metadata.
+	err = rt.Set("assert_has_trace_id_metadata", func(expected bool, expectedTraceID string) {
+		gotTraceID, hasTraceID := testCase.TestRuntime.VU.State().Tags.GetCurrentValues().Metadata["trace_id"]
+		require.Equal(t, expected, hasTraceID)
+
+		if expectedTraceID != "" {
+			require.Equal(t, expectedTraceID, gotTraceID)
+		}
+	})
+	require.NoError(t, err)
+
+	// Assert there is no trace_id key in vu metadata before calling an instrumented
+	// function, and that it's cleaned up after the call.
+	t.Cleanup(testCase.TestRuntime.EventLoop.WaitOnRegistered)
+	err = testCase.TestRuntime.EventLoop.Start(func() error {
+		_, err = rt.RunString(httpBin.Replacer.Replace(`
+        	assert_has_trace_id_metadata(false)
+        	http.request("GET", "HTTPBIN_URL")
+        	assert_has_trace_id_metadata(false)
+        `))
+
+		return err
+	})
+	require.NoError(t, err)
+	close(samples)
+
+	var sampleRead bool
+	for sampleContainer := range samples {
+		for _, sample := range sampleContainer.GetSamples() {
+			require.NotEmpty(t, sample.Metadata["trace_id"])
```

Comment:
Here again you could check what the value is, given that it is deterministc, but you don't